tests/manage/pv_services/pvc_clone/test_pvc_to_pvc_clone.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import pytest
+
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import (
+    skipif_ocs_version, ManageTest, tier1
+)
+from ocs_ci.ocs.resources import pod
+from tests import helpers
+
+logger = logging.getLogger(__name__)
+
+
+@tier1
+@skipif_ocs_version('<4.6')
+@pytest.mark.parametrize(
+    argnames=["interface_type"],
+    argvalues=[
+        pytest.param(
+            constants.CEPHBLOCKPOOL, marks=pytest.mark.polarion_id("")
+        )
+    ]
+)
+class TestPvcToPvcClone(ManageTest):
+    """
+    Tests to verify PVC to PVC clone feature
+    """
+    @pytest.fixture(autouse=True)
+    def setup(self, interface_type, storageclass_factory, pvc_factory, pod_factory):
+        """
+        create resources for the test
+        Args:
+            interface_type(str): The type of the interface
+                (e.g. CephBlockPool, CephFileSystem)
+            storageclass_factory: A fixture to creare new storage class
+            pvc_factory: A fixture to create new pvc
+            pod_factory: A fixture to create new pod
+        """
+        self.sc_obj = storageclass_factory(interface=interface_type)
```

Comment:
@jilju , since both the pvc and the cloned pvc need to be of the same storage class, Is it alright to create one and then use for both(It also highlights their requirement of being of same storage class)? ... Any thoughts?

Reply:
If SC is not passed to pvc_factory, ocs-storagecluster-ceph-rbd will be used by default. Here self.sc_obj is created but not passed to pvc_factory. So pvc_factory will use ocs-storagecluster-ceph-rbd storage class. This same storage class can be used for clone. But self.sc_obj SC is passed to helpers.create_pvc_clone. So it will try to create clone using a different SC than parent. 

I feel the create_pvc_clone function should be moved to class PVC in ocs-ci/ocs_ci/ocs/resources/pvc.py.

```
class PVC(OCS):
   <rest of the code>

   def create_pvc_clone(pvc_name, volume_mode):

       pvc_data = templating.load_yaml(constants.CSI_RBD_PVCCLONE_YAML)
       pvc_data['metadata']['name'] = (
            pvc_name if pvc_name else create_unique_resource_name(
            '    cloned', 'pvc'
            )
        )
        pvc_data['spec']['storageClassName'] = self.backed_sc
        if volume_mode:
            pvc_data['spec']['volumeMode'] = volume_mode
        pvc_data['spec']['dataSource']['name'] = self.name
        # TODO: Get access mode using self.get_pvc_access_mode and set the value in pvc_data
        # TODO: Get size in Gi using self.size and set the value in pvc_data
        ocs_obj = pvc.PVC(**pvc_data)
        created_pvc = ocs_obj.create(do_reload=do_reload)
        assert created_pvc, f"Failed to create resource {pvc_name}"
        return ocs_obj

    <rest of the code>
```